### PR TITLE
Fixed zooming bug in native-host-ext [Chrome]

### PR DESCRIPTION
**Pull request by mfmarlow**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description
Fixed a bug that misplaced the attached window when zooming on Chrome. Now gets pixel ratio data from the extension to determine the proper x location of chat. Still supports extension versions 1.2 and 1.3. 

Requires: https://github.com/Chatterino/chatterino-browser-ext/pull/28

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## Unversioned
 
+- Bugfix: Fixed bug misplacing chat when zooming on Chrome with Chatterino Native Host extension (#1936)
 - Minor: Disable checking for updates on unsupported platforms (#1874)
 - Bugfix: Fix bug preventing users from setting the highlight color of the second entry in the "User" highlights tab (#1898)
 - Bugfix: /usercard command will now respect the "Automatically close user popup" setting (#1918)
```
Comment (alazymeme) [inline CHANGELOG.md:8]:
```suggestion
- Minor: Disable checking for updates on unsupported platforms (#1874)
- Bugfix: Fixed bug misplacing chat when zooming on Chrome with Chatterino Native Host extension (#1936)
- Bugfix: Fix bug preventing users from setting the highlight color of the second entry in the "User" highlights tab (#1898)
- Bugfix: /usercard command will now respect the "Automatically close user popup" setting (#1918)
```

**Review by alazymeme — CHANGES_REQUESTED** [review]
Your PR introduces a new issue to where the chatterino extension will load with the page and not minimize at all despite there being no chat there.

image

File: src/singletons/NativeMessaging.cpp [hunk 2]
```diff
@@ -198,12 +197,15 @@ void NativeMessagingServer::ReceiverThread::handleMessage(
         AttachedWindow::GetArgs args;
         args.winId = root.value("winId").toString();
         args.yOffset = root.value("yOffset").toInt(-1);
-        args.x = root.value("size").toObject().value("x").toInt(-1);
+        args.x = root.value("size").toObject().value("x").toDouble(-1.0);
+        args.pixelRatio =
+            root.value("size").toObject().value("pixelRatio").toDouble(-1.0);
         args.width = root.value("size").toObject().value("width").toInt(-1);
         args.height = root.value("size").toObject().value("height").toInt(-1);
```
Comment (23rd) [inline src/singletons/NativeMessaging.cpp:204]:
```suggestion

        {
            const auto sizeObject = root.value("size").toObject();
            args.x = sizeObject.value("x").toDouble(-1.0);
            args.pixelRatio = sizeObject.value("pixelRatio").toDouble(-1.0);
            args.width = sizeObject.value("width").toInt(-1);
            args.height = sizeObject.value("height").toInt(-1);
        }

```

**Comment by fourtf** [comment]
Can someone verify that this works?

**Comment by alazymeme** [comment]
> Can someone verify that this works? Also check with hi-dpi enabled in windows.

I can check, just need some form of a comment or change in regards to https://github.com/Chatterino/chatterino2/pull/1936#discussion_r494196666 and then I'll test it again

**Comment by jammehcow** [comment]
Is this fix still needed or can it be closed?

**Review by ALazyMeme — APPROVED** [review]

**Comment by ALazyMeme** [comment]
Confirmed the new chatterino build from this pr works just fine with the normal extension (without the zoom-related updated).

**Comment by ALazyMeme** [comment]
@pajlada merge pls

**Review by pajlada — APPROVED** [review]

**Review by pajlada — APPROVED** [review]

**Pull request merged**